src/gobprepare/prepare_client.py

```diff
@@ -254,6 +259,58 @@ def action_check_row_counts(self, action: RowCountConfig) -> bool:
                 )
         return tables_counts_ok
 
+    def action_check_source_sync_complete(self, action: SyncSchemaConfig) -> bool:
+        """Check source sync action. Verify when sync tables from databricks is completed.
+
+        :param action:
+        :return:
+        """
+        table_name = action["table_name"]
+        schema = action["schema"]
+
+        if not self._dst_datastore.query(check_table_existence_query("public", table_name)):
+            raise GOBException(f"Table 'public.{table_name}' does not exists.")
+
+        self._check_last_schema_sync(table_name, schema)
+        self._update_sync_schema(table_name, schema, update_action = 'start_prepare')
+        return True
+
+    def action_complete_prepare(self, action: SyncSchemaConfig) -> bool:
+        """update sync table after prepare is completed.
+
+        :param action:
+        :return:
+        """
+        table_name = action["table_name"]
+        schema = action["schema"]
+        self._check_last_schema_sync(table_name, schema)
+        self._update_sync_schema(table_name, schema, update_action = 'end_prepare')
+        return True
+
+    def _check_last_schema_sync(self, table_name, schema) -> None:
+        select_query = f"SELECT * FROM public.{table_name} WHERE sync_schema = '{schema}'"
+        try:
+            schema_sync_data = next(self._dst_datastore.query(select_query))
+        except StopIteration:
+            raise GOBException(f"No record for schema '{schema}' found in 'public.synced_schemas'.")
+        if schema_sync_data[1] is None:
```

Comment:
Looking much better! One comment though:
If you make it a "SELECT column_name ...", then you can use schema_sync_data[0]. Now it's not really clear which column you're quering.

Reply:
Idd. Dit zijn de wijzigingen die ik van plan om toe te voegen.
Ik haal nu twee fielden `last_sync_end`, `last_sync_start` op want de `last_sync_start` heb ik nodig in de logging als het fout gaat.